### DISCUSSION TITLE
Atom injection

### DISCUSSION THREAD

**Discussion by kirrosh** [pmndrs/jotai#893] [Show and tell] [answered]
Hi guys, what do you think about this?

Range component with atom injection:
![image](https://user-images.githubusercontent.com/4011023/146268944-1ab90a12-524c-47fd-b720-bbed30e7ae03.png)

Usage

![image](https://user-images.githubusercontent.com/4011023/146269258-4bf2a119-b316-42e5-b503-5eff6ab3b2f1.png)

Now **Atom** can be used anywhere and we don't need save state in parent component. 
**Also** there is no Context so this component can be reused without creating a container, we only need to create atom outside of **React**.

**Comment by dai-shi** [comment, marked as answer]
Yes, passing an atom config in component props is a very good practice. We should improve the docs to promote it.

You might want to use `useMemo` in `useAtom`: `useAtom(useMemo(() => atom(...), [rangeAtom]))`
otherwise, it could cause an infinite loop.

**Reply by kirrosh** [reply]
I tried both and found no difference, so thanks a lot for your advice :)
